lib/revocation.js

```diff
@@ -0,0 +1,135 @@
+var assert = require('assert')
+    , async = require('async')
+    , aws = require('aws-sdk')
+    , LRU = require('lru-cache')
+    ;
+
+var dynamo;
+var lru; // LRU cache of token revocation information
+
+exports.init = function (options) {
```

Comment:
There 2 code paths to reach this from within the proxy, but always with the same options. It is not possible to predict which code path will get called first.